src/hooks/useWatchPendingTxs.ts

```diff
@@ -110,58 +84,7 @@ export const useWatchPendingTransactions = ({ address }: { address: string }) =>
       }
       return updatedTransaction;
     },
-    [address, processSupportedNetworkTransaction, processFlashbotsTransaction, refreshAssets]
-  );
-
-  const processNonces = useCallback(
-    (txs: RainbowTransaction[]) => {
-      const userTxs = txs.filter(tx => address?.toLowerCase() === tx.from?.toLowerCase());
-      const chainIds = [
-        ...new Set(
-          userTxs.reduce((acc, tx) => {
-            acc.add(tx.chainId);
-            return acc;
-          }, new Set<ChainId>())
-        ),
-      ];
-      let flashbotsTxFailed = false;
-      const highestNoncePerChainId = userTxs.reduce((acc, tx) => {
-        // if tx is not on mainnet, we don't care about the nonce
-        if (tx.chainId !== ChainId.mainnet) {
-          acc.set(tx.chainId, tx.nonce);
-          return acc;
-        }
-        // if tx is flashbots and failed, we want to use the lowest nonce
-        if (tx.flashbots && tx?.flashbotsStatus === FlashbotsStatus.FAILED && tx?.nonce) {
-          // if we already have a failed flashbots tx, we want to use the lowest nonce
-          if (flashbotsTxFailed && tx.nonce < acc.get(tx.chainId)) {
-            acc.set(tx.chainId, tx.nonce);
-          } else {
-            acc.set(tx.chainId, tx.nonce);
-            flashbotsTxFailed = true;
-          }
-          // if tx succeeded, we want to use the highest nonce
-        } else if (!flashbotsTxFailed && tx?.nonce && tx.nonce > acc.get(tx.chainId)) {
-          acc.set(tx.chainId, tx.nonce);
-        }
-        return acc;
-      }, new Map());
-
-      chainIds.map(async chainId => {
-        const provider = getProvider({ chainId });
-        const providerTransactionCount = await provider.getTransactionCount(address, 'latest');
-        const currentProviderNonce = providerTransactionCount - 1;
-        const currentNonceForChainId = highestNoncePerChainId.get(chainId) - 1;
-
-        setNonce({
-          address,
-          chainId,
-          currentNonce: currentProviderNonce > currentNonceForChainId ? currentProviderNonce : currentNonceForChainId,
```

Comment:
this `processNonces` function is no longer needed - we don't use `latestConfirmedNonce` and the `currentNonce` will have been made up to date already when adding / updating the pending txn